plugins/kubernetes/app/models/kubernetes/role.rb

```
@@ -19,6 +19,37 @@ class Role < ActiveRecord::Base
 
     scope :not_deleted, -> { where(deleted_at: nil) }
 
+    # create initial roles for a project by reading kubernetes/*{.yml,.yaml,json} files into roles
+    def self.seed!(project, git_ref)
+      kubernetes_config_files_in_repo(project, git_ref).each do |config_file|
+        create!(
+          project: project,
+          config_file: config_file.file_path,
+          name: config_file.deployment.metadata.labels.role,
+          service_name: config_file.service && config_file.service.metadata.name,
+          ram: config_file.deployment.ram_mi, # TODO: remove this column
+          cpu: config_file.deployment.cpu_m, # TODO: remove this column
+          replicas: config_file.deployment.spec.replicas || 1, # TODO: remove this column
+          deploy_strategy: config_file.deployment.strategy_type
+        )
+      end
+    end
+
+    def self.for_deploy(project, git_sha)
```

Should raise an error if `git_sha` is not a valid ref for that project.